pkg/fileutil/fileutil.go

```diff
@@ -121,7 +123,18 @@ func CopyTree(src, dest string) error {
 			return fmt.Errorf("unsupported mode: %v", mode)
 		}
 
-		if err := os.Lchown(target, int(info.Sys().(*syscall.Stat_t).Uid), int(info.Sys().(*syscall.Stat_t).Gid)); err != nil {
+		var srcUid = info.Sys().(*syscall.Stat_t).Uid
+		var srcGid = info.Sys().(*syscall.Stat_t).Gid
+
+		if uidRange.Count > 0 && (srcUid >= uidRange.Count || srcGid >= uidRange.Count) {
```

Comment:
I suggest putting count and shift checks to a member function of uid.Range. Something like:
```go
func (r *UidRange) Shift(uint32 uid, uint32 gid) (uint32, uint32, error) {
    if r.Count > 0 && (uid >= r.Count || gid >= r.Count) {
        return 0, 0, fmt.Errorf("uid %d or gid %d are out of range %d", uid, gid, r.Count)
    }
    shiftedUid, shiftedGid, idMax := uid + r.Shift, gid + r.Shift, 0xFFFFFFFF
    if r.Shift > 0 && (shiftedUid >= idMax || shiftedGid >= idMax) {
        return 0, 0, fmt.Errorf("uid %d or gid %d are out of range %d after shifting", shiftedUid, shiftedGid, idMax)
    }
    return shiftedUid, shiftedGid, nil
```
You can copy and paste this code. ;)